src/observable/GenerateObservable.ts

```diff
@@ -37,16 +60,73 @@ export class GenerateObservable<T, S> extends Observable<T> {
       super();
   }
 
+  /**
+   * Generates an observable sequence by running a state-driven loop
+   * producing the sequence's elements, using the specified scheduler
+   * to send out observer messages.
+   *
+   * @example <caption>Produces sequence of 0, 1, 2, ... 9, then completes.</caption>
+   * var res = Rx.Observable.generate(0, x => x < 10, x => x + 1, x => x);
+   * 
+   * @example <caption>Using asap scheduler, produces sequence of 2, 3, 5, then completes.</caption>
+   * var res = Rx.Observable.generate(1, x => x < 5, x => x * 2, x => x + 1, Rx.Scheduler.asap);
+   *
+   * @param {Mixed} initialState Initial state.
+   * @param {Function} condition Condition to terminate generation (upon returning false).
+   * @param {Function} iterate Iteration step function.
+   * @param {Function} resultSelector Selector function for results produced in the sequence.
+   * @param {Scheduler} scheduler Scheduler on which to run the generator loop. If not provided, defaults to emit immediately.
+   * @returns {Observable} The generated sequence.
+   */
```

Comment:
The structure we use in other operators is (in this order):
- 2 line short technical description of the operator.
- 2 line short *informal* description (with plain English, for dummies) of the operator.
- `<img>` with the marble diagram
- Long technical description of the operator, in details and explaining all the optional arguments too.
- Examples (yours are good already :+1:)
- `@see` linking to similar operators. E.g. I would link to `from` and `create`.
- Params (yours is good, but `Mixed` is not a valid type)

It's easy for me to do a separate PR for this, if you prefer.